File: samples/C# Samples/TheMinimumSystem/TheMinimumSystem/SourceListForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace TheMinimumSystem
{
    public partial class SourceListForm : Form
    {
        private List<string> m_ListSourceNames = null;
        private int m_SelectedIndex = -1;
        public SourceListForm(List<string> listSourceNames)
        {
            InitializeComponent();
            m_ListSourceNames = listSourceNames;
            if (m_ListSourceNames == null)
                button1.Enabled = false;

            foreach(string temp in m_ListSourceNames)
            {
                listBox1.Items.Add(temp);
            }
            listBox1.SelectedIndex = m_ListSourceNames.Count - 1;
        }

        private void button1_Click(object sender, EventArgs e)
        {
            m_SelectedIndex = listBox1.SelectedIndex;
            this.Close();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            this.Close();

        }

        public int GetSelectedIndex()
        {
            return m_SelectedIndex;
        }
    }
}

File: samples/VB .NET Samples/TheMinimumSystem/TheMinimumSystem/SourceListForm.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Public Partial Class SourceListForm
	Inherits Form
	Private m_ListSourceNames As List(Of String) = Nothing
	Private m_SelectedIndex As Integer = -1
	Public Sub New(listSourceNames As List(Of String))
		InitializeComponent()
		m_ListSourceNames = listSourceNames
		If m_ListSourceNames Is Nothing Then
			button1.Enabled = False
		End If

		For Each temp As String In m_ListSourceNames
			listBox1.Items.Add(temp)
		Next
		listBox1.SelectedIndex = m_ListSourceNames.Count - 1
	End Sub

    Private Sub button1_Click(sender As Object, e As EventArgs) Handles button1.Click
        m_SelectedIndex = listBox1.SelectedIndex
        Me.Close()
    End Sub

    Private Sub button2_Click(sender As Object, e As EventArgs) Handles button2.Click
        Me.Close()

    End Sub

	Public Function GetSelectedIndex() As Integer
		Return m_SelectedIndex
	End Function
End Class
